packages/rspack/src/plugins/utils/apply-base-config.ts

```diff
@@ -0,0 +1,418 @@
+            },
+          };
+        }),
+      })
+    );
+  }
+  if (options.generatePackageJson && executorContext) {
+    plugins.push(new GeneratePackageJsonPlugin({ ...options, tsConfig }));
+  }
+
+  if (options.statsJson) {
+    plugins.push(new StatsJsonPlugin());
+  }
+
+  const externals = [];
+  if (options.target === 'node' && options.externalDependencies === 'all') {
+    const modulesDir = `${options.root}/node_modules`;
+    externals.push(nodeExternals({ modulesDir }));
+  } else if (Array.isArray(options.externalDependencies)) {
+    externals.push(function (ctx, callback: Function) {
+      if (options.externalDependencies.includes(ctx.request)) {
+        // not bundled
+        return callback(null, `commonjs ${ctx.request}`);
+      }
+      // bundled
+      callback();
+    });
+  }
+
+  config.resolve = {
+    ...config.resolve,
+    extensions: [...(config?.resolve?.extensions ?? []), ...extensions],
+    alias: {
+      ...(config.resolve?.alias ?? {}),
+      ...(options.fileReplacements?.reduce(
+        (aliases, replacement) => ({
+          ...aliases,
+          [replacement.replace]: replacement.with,
+        }),
+        {}
+      ) ?? {}),
+    },
+    mainFields: config.resolve?.mainFields ?? mainFields,
+  };
+
+  config.externals = externals;
+
+  // Enabled for performance
+  config.cache = true;
+  config.module = {
+    ...config.module,
+    rules: [
+      ...(config?.module?.rules ?? []),
+      options.sourceMap && {
+        test: /\.js$/,
+        enforce: 'pre' as const,
+        loader: require.resolve('source-map-loader'),
+      },
+      {
+        // There's an issue resolving paths without fully specified extensions
+        // See: https://github.com/graphql/graphql-js/issues/2721
+        // TODO(jack): Add a flag to turn this option on like Next.js does via experimental flag.
+        // See: https://github.com/vercel/next.js/pull/29880
+        test: /\.m?jsx?$/,
+        resolve: {
+          fullySpecified: false,
+        },
+      },
+      // There's an issue when using buildable libs and .js files (instead of .ts files),
+      // where the wrong type is used (commonjs vs esm) resulting in export-imports throwing errors.
+      // See: https://github.com/nrwl/nx/issues/10990
+      {
+        test: /\.js$/,
+        type: 'javascript/auto',
+      },
+      // Rspack's docs only suggest swc for TS compilation
+      //https://rspack.dev/guide/tech/typescript
+      {
+        test: /\.([jt])sx?$/,
+        loader: 'builtin:swc-loader',
+        exclude: /node_modules/,
+        options: {
+          jsc: {
+            parser: {
+              syntax: 'typescript',
+              decorators: true,
+              tsx: true,
+            },
+            transform: {
+              react: {
+                pragma: 'React.createElement',
+                pragmaFrag: 'React.Fragment',
+                throwIfNamespace: true,
+                development: false,
```

Comment:
Before, I think this was determined based on `NODE_ENV` or `options.mode` should this still be the case?

Reply:
Hmm 🤔 good spot, yes likely